portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -1,38 +1,86 @@
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.Optional;
 
 @WebServlet("/login")
 public class HomeServlet extends HttpServlet {
   private static final String HOME_PATH = "/";
+  private static final String USER_INFO = "userInfo";
+  private static final String ID = "id";
+  private static final String COMMENT_NICKNAME = "nickname";
+  // Hard-coded email of webpage admin.
+  private static final String ADMIN_EMAIL = "amytn@google.com";
+  private static final String NICKNAME_SERVLET = "/nickname";
+
+  private static String userEmail;
```

Comment:
I am a little confused as to why these are static if non-static methods can change them. Can't there be more than 1 nickname/userEmail based on who logged in?

Reply:
There should only be one nickname/email for each user. Since there is only 1 current user, there would only be 1 nickname/email per session. The nickname can be updated multiple times but there should still be only 1. 

I added a comment so hopefully, it makes more sense.